src/indexedtable.jl

```diff
@@ -101,10 +101,10 @@ function table(::Val{:serial}, cols::Tup;
                 cs = permute!(cs, perm)
             end
         elseif copy
-            cs = Base.copy(cs)
+            cs = copyto!(similar(cs), cs)
         end
     elseif copy
-        cs = Base.copy(cs)
+        cs = copyto!(similar(cs), cs)
```

Comment:
Curious why this change is necessary? Shouldn't `copy` work?

Reply:
Using `copy` would create a subtle bug. This code expects `similar` and `copy` to return things of the same value (for `index` and `index_buffer` for example). However this is in general false for arrays. `copy(1:3) isa UnitRange` whereas `similar(1:3) isa Array`. This used to work with `Columns` because `Columns` has no custom method for `copy`, so it would default to `copymutable(x) = copyto!(similar(x), x)` which would in turn make all columns mutable. `copy` for a `StructArray` is instead defined as column-wise copy (which I think is correct). 

I think `copymutable` is the correct choice here in general as we want to be able to modify the indexing arrays as well as the buffer arrays, but I don't think copying a `StructArray` should in general make the field arrays mutable.